### DISCUSSION TITLE
interrupt handlers in Forth?

### DISCUSSION THREAD

**Discussion by cminter** [jkotlinski/durexforth#374] [Ideas]
Is it possible now to do interrupt handlers in Forth?

I tried to look, but I don't understand yet enough about how the implementation works enough to even recognize whether it is there. :)

FYI:  I read an old article '[Zero Overhead Forth Interrupts](http://6502.org/tutorials/zero_overhead_forth_interrupts.html)' [[6502.org](http://6502.org)]

**Comment by jkotlinski** [comment]
There is no support for writing interrupt handlers in pure Forth. What would need to be done, roughly, is:

 * Adding a word to set the CIA/VIC/NMI interrupt vectors
 * Adding a word to save the context (a/x/y flags, w/w2/w3 content) to return stack when entering the interrupt handler
 * Adding words to clear the triggering CIA/VIC interrupt flags, restore context and return from the interrupt handler through RETI.

It would not be a terrible lot of work, however it is anyway not really a good idea to write interrupt handlers in Forth. The 6502 is very slow and interrupt handlers need to be as fast as possible, at least for demo programming, so they are better written in assembler.

**Comment by Whammo** [comment]
Check out #443
